services/provider/server/server.go

```diff
@@ -593,6 +593,12 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 				return nil, status.Errorf(codes.Internal, "failed to get %s CephBlockPoolRadosNamespace. %v", cephRes.Name, err)
 			}
 
+			cbp := &rookCephv1.CephBlockPool{}
+			err = s.client.Get(ctx, types.NamespacedName{Name: cephRes.Name, Namespace: s.namespace}, rns)
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "failed to get %s CephBlockPool. %v", cephRes.Name, err)
+			}
+
```

Comment:
Not sure what you are trying to achieve here, you are creating a cpb var but you are getting the radosnamespace. 
Also, we should not depend on one cephBlockPool to decide if we are enabling mirroring or not. There might be a case where we might need to blacklist some blockpools for mirroring.

Reply:
No, I'm tagging the CBPRN class with a flag that will signal to StorageClaim's controller whether mirroring's enabled for said class. 

> There might be a case where we might need to blacklist some blockpools for mirroring.

Care to elaborate? What indicates a blacklisted pool?

Reply:
```
err = s.client.Get(ctx, types.NamespacedName{Name: cephRes.Name, Namespace: s.namespace}, rns)
```
You are getting the rns here, not the cbl, is this intended?

Reply:
Discussed over DMs:

- CBP is the only source to fetch the mirroring flag from.
- In case of blacklisted pools we can have their flag indicating that override the mirroring flag.